go/runtime/bundle/bundle.go

```diff
@@ -177,7 +184,16 @@ func (bnd *Bundle) MrSigner(id component.ID) (*sgx.MrSigner, error) {
 		mrSigner = sgx.FortanixDummyMrSigner
 	default:
 		// Load the actual signature.
-		sigPk, _, err := sigstruct.Verify(bnd.Data[comp.SGX.Signature])
+		d, ok := bnd.Data[comp.SGX.Signature]
+		if !ok {
+			return nil, fmt.Errorf("runtime/bundle: no SGX signature for '%s'", id)
+		}
+		b, err := ReadAllData(d)
```

Comment:
What do you mean? This only reads the signature data as `d` is just the `bnd.Data[comp.SGX.Signature]` file, not everything else?

Reply:
In general `bnd.Data[comp.SGX.Signature]` is `Data` which could contain a lot of bytes. Since we expect to read signature with a fixed size, one could stop reading when `sigstructSize` bytes are read and throw error if EOF was not reached. This is a small optimization in case signature is invalid and has size in GB. 
As said, this are further optimizations, probably not worth of our time.

Reply:
I see, yeah once we start dealing with potentially malicious bundles, we should be more careful (in general).